src/daemon/server.py

```diff
@@ -0,0 +1,487 @@
+import asyncio
+import json
+import logging
+import os
+import signal
+import subprocess
+import sys
+import traceback
+from typing import Dict, Any
+from sys import platform
+from aiohttp import web
+
+
+import websockets
+from src.cmds.init import chia_init
+
+from src.util.ws_message import format_response
+from src.util.json_util import dict_to_json_str
+
+try:
+    import fcntl
+
+    has_fcntl = True
+except ImportError:
+    has_fcntl = False
+
+from src.util.config import load_config
+from src.util.logging import initialize_logging
+from src.util.path import mkdir
+from src.util.service_groups import validate_service
+
+log = logging.getLogger(__name__)
+
+# determine if application is a script file or frozen exe
+if getattr(sys, "frozen", False):
+    name_map = {
+        "chia": "chia",
+        "chia-check-plots": "check_plots",
+        "chia-create-plots": "create_plots",
+        "chia-wallet": "websocket_server",
+        "chia_full_node": "start_full_node",
+        "chia_harvester": "start_harvester",
+        "chia_farmer": "start_farmer",
+        "chia_introducer": "start_introducer",
+        "chia_timelord": "start_timelord",
+        "chia_timelord_launcher": "timelord_launcher",
+        "chia_full_node_simulator": "start_simulator",
+        "plotter": "create_plots",
+    }
+
+    def executable_for_service(service_name):
+        application_path = os.path.dirname(sys.executable)
+        if platform == "win32" or platform == "cygwin":
+            executable = name_map[service_name]
+            path = f"{application_path}/{executable}.exe"
+            return path
+        else:
+            path = f"{application_path}/{name_map[service_name]}"
+            return path
+
+
+else:
+    application_path = os.path.dirname(__file__)
+
+    def executable_for_service(service_name):
+        return service_name
+
+
+class WebSocketServer:
+    def __init__(self, root_path):
+        self.root_path = root_path
+        self.log = log
+        self.services: Dict = dict()
+        self.connections: Dict[str, Any] = dict()  # service_name : WebSocket
+        self.remote_address_map: Dict[str, str] = dict()  # remote_address: service_name
+        self.ping_job = None
+
+    async def start(self):
+        self.log.info("Starting Daemon Server")
+
+        def master_close_cb():
+            asyncio.ensure_future(self.stop())
+
+        try:
+            asyncio.get_running_loop().add_signal_handler(
+                signal.SIGINT, master_close_cb
+            )
+            asyncio.get_running_loop().add_signal_handler(
+                signal.SIGTERM, master_close_cb
+            )
+        except NotImplementedError:
+            self.log.info("Not implemented")
+
+        self.websocket_server = await websockets.serve(
+            self.safe_handle, "localhost", 55400
```

Comment:
This will cause issues running from docker containers due to this issue https://github.com/Chia-Network/chia-blockchain/issues/219

Happy to workaround for now if you don't want to delay the release.

Reply:
I think you're going to still going to have to work around this in docker for 1.6 and we'll try to fix for 1.7.